### PR TITLE
DMP-4107 Enhance auditing for admin task automation activities

### PR DESCRIPTION
**Pull request by Ben-Edwards-cgi**
### Change description ###
# Summary of Changes

The recent updates to the codebase enhance the auditing functionality by allowing additional data to be recorded alongside audit activities. This improvement ensures that more context is captured when specific tasks are executed, thus improving traceability and accountability.

## Highlights

- **New Feature**: Added the ability to record additional data with audit activities.
- **Test Updates**: Adjusted unit tests to verify that the new additional data is correctly recorded during task executions.
  - For example, when a task is run manually, it now logs the name of the task ("ProcessDailyList").
  - or when a task is disabled/enabled, it now logs the name of the task and its disabled/enabled status i.e "ProcessDailyList disabled" / "ProcessDailyList enabled".
  - When updating task status, it logs whether the task is enabled or disabled.
- **Audit API Changes**: Enhanced the `AuditApi` interface to accept an additional string parameter for extra information.
- **Implementation Changes**: The implementation classes were updated to support the new method for recording audits with additional data.
- **Transactional Integrity**: Ensured that the new audit recording methods maintain transactional behavior, preserving data consistency.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/audit/service/AuditService.java [hunk 1]
```diff
@@ -8,4 +9,6 @@ public interface AuditService {
 
     void recordAudit(AuditActivity activity, UserAccountEntity userAccountEntity, CourtCaseEntity courtCase);
 
+    @Transactional
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/darts/audit/service/AuditService.java:12]: No need for `@Transactional` on the interface.
 The Spring team's recommendation is that you only annotate concrete classes with the @Transactional annotation, as opposed to annotating interfaces

File: src/main/java/uk/gov/hmcts/darts/task/service/impl/AdminAutomatedTasksServiceImpl.java [hunk 2]
```diff
@@ -83,7 +83,7 @@ public DetailedAutomatedTask updateAutomatedTask(Integer taskId, AutomatedTaskPa
 
         if (automatedTaskPatch.getIsActive() != null) {
             automatedTask.setTaskEnabled(automatedTaskPatch.getIsActive());
-            auditApi.record(ENABLE_DISABLE_JOB);
+            auditApi.record(ENABLE_DISABLE_JOB, automatedTask.getTaskName() + " " + (automatedTaskPatch.getIsActive() ? "enabled" : "disabled"));
         }
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/darts/task/service/impl/AdminAutomatedTasksServiceImpl.java:87]: the ticket is asking also to add a log:
```
Also, add a log message to the enable/disable job part, in the `AdminAutomatedTasksServiceImpl` class updateAutomatedTask method, making it clear what task was changed and if enabled or disabled.
```

File: src/main/java/uk/gov/hmcts/darts/audit/service/AuditService.java (hunk 1 above)
Comment (Ben-Edwards-cgi) [reply src/main/java/uk/gov/hmcts/darts/audit/service/AuditService.java:12]: Agreed updated :) (I think this was just inteliJ auto adding it :D)

File: src/main/java/uk/gov/hmcts/darts/task/service/impl/AdminAutomatedTasksServiceImpl.java (hunk 2 above)
Comment (Ben-Edwards-cgi) [reply src/main/java/uk/gov/hmcts/darts/task/service/impl/AdminAutomatedTasksServiceImpl.java:88]: Nice spot updated

**Pull request merged**